### PR TITLE
Update federation gem

### PR DESCRIPTION
**Pull request by SuperTux88**
- use gem rails-engine controller for receive (creates sidekiq jobs over callbacks)
- rewrite federation-tests
- add subscribe-url to webfinger
- fix federation-factories with spring

### CODE REVIEW

File: spec/integration/federation/shared_receive_retraction.rb [hunk 1]
```diff
@@ -1,43 +1,62 @@
...
-    Workers::ReceiveEncryptedSalmon.new.perform(@user.id, generate_retraction(entity_name, target_object))
-
-    expect(target_klass.exists?(guid: target_object.guid)).to be(false)
+def mock_private_keys_for_retraction(entity_name, entity, sender)
+  if %i(signed_retraction_entity relayable_retraction_entity).include?(entity_name)
+    allow(DiasporaFederation.callbacks).to receive(:trigger)
+                                             .with(:fetch_private_key_by_diaspora_id, sender.diaspora_handle)
+                                             .and_return(sender.encryption_key)
   end
+  if entity_name == :relayable_retraction_entity
```
Comment (diaspora-code-review) [inline spec/integration/federation/shared_receive_retraction.rb]: Use a guard clause instead of wrapping the code inside a conditional expression.

**Comment by denschub** [comment]
I've merged this, thank you very much!

**Comment by cmrd-senya** [comment]
Oh no, this completely breaks my #6595 :'(

**Comment by cmrd-senya** [comment]
I'll have to rewrite everything :(

**Comment by cmrd-senya** [comment]
Why not discuss it with me as long as I participate to that as well? @SuperTux88, I've been pinging you on IRC for several days to sort it out.

**Comment by cmrd-senya** [comment]
I'll revert 5392c6e6a905552c8b0831dd7983f7511ca5c622 and 24f5244f766d1c7c0d277b40c52724239bbebb12 in my PR, so the changes apply.

**Comment by denschub** [comment]
Uhm? Reverting code we just wrote is not really an option... We had the discussion on the location of our glue code before and neither I nor @SuperTux88 think it's needed to spread the glue code unnecessarily. You have to add some workarounds in your test just like SuperTux88 did here, but writing slightly more complicated tests is better than writing more complicated app source...

If you have issues with adopting your code, please ping us on IRC. Again, reverting code we just added is not an option since we usually don't just write code for fun... :)

**Comment by cmrd-senya** [comment]
Well, @SuperTux88 changed my tests and he changed an outdated version while a new version was available in the PR. I was trying to discuss it with him and he didn't reply neither on my PR for diaspora_federation, nor on my IRC requests. And now I find that he has introduced changes to the code that I have overriden by a new PR. I think it is fair enough, if I revert his changes and he applies his refactorings again against the up-to-date version. Otherwise, fuck that all, I don't like to work like that. I don't just write code for fun either and I dislike my work being just messed up.

**Comment by cmrd-senya** [comment]
My code was ready for merge for more than a week. The only thing I waited for to remove [WIP] status was some patches included in the release of diaspora_federation. Instead, I got mess up of my work.

**Comment by SuperTux88** [comment]
Sorry, I didn't had much time for IRC the last days.

Maybe you can revert 5392c6e6a905552c8b0831dd7983f7511ca5c622 if you can write the tests that you don't need to revert 24f5244f766d1c7c0d277b40c52724239bbebb12 (I discussed this with @jhass and @denschub month ago, that we want all glue-code at one place, and I proved with this PR that it isn't needed to move this code out of the initializer to write tests). But I also think that some refactorings are better: don't use global variables/methods everywhere.

**Comment by cmrd-senya** [comment]
The tests' mocking relies on Person.hcard and Person.webfinger methods I introduced. I don't argue for moving glue-code outside. But I wasn't informed on that desicion when the previous tests PR was merged. You changed that in the refactorings and I'm sure your refactorings are worthy and useful. Could you just please reapply them on top of #6595 then? We can then move out the glue code. Otherwise, if you enumerate what to fix in the #6595, I can check it myself.

Anyway, I need this little patch for new tests to work - https://github.com/SuperTux88/diaspora_federation/pull/24.

**Pull request closed without merging**